include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf.h

```diff
@@ -132,6 +140,39 @@ class TimeIntBDF : public TimeIntBDFBase<Number>
   std::shared_ptr<HelpersALE<Number> const> helpers_ale;
 
 private:
+  template<typename Lambda>
+  void
+  get_quantity_and_times(std::vector<VectorType const *> & quantities,
+                         std::vector<double> &             times,
+                         Lambda const &                    get_quantity) const
+  {
+    unsigned int const current_order = this->get_current_order();
+
+    quantities.resize(current_order);
+    this->fill_at_previous_times(quantities.begin(), quantities.end(), get_quantity);
+
+    times.resize(current_order);
+    this->fill_with_previous_times(times.begin(), times.end());
+  }
+
+  template<typename Lambda1, typename Lambda2>
+  void
+  get_quantities_and_times_np(std::vector<VectorType const *> & quantities,
+                              std::vector<double> &             times,
+                              Lambda1 const &                   get_quantity,
+                              Lambda2 const &                   get_quantity_np) const
+  {
+    unsigned int const current_order = this->get_current_order();
+
+    quantities.resize(current_order + 1);
+    quantities[0] = get_quantity_np();
+    this->fill_at_previous_times(quantities.begin() + 1, quantities.end(), get_quantity);
```

Comment:
Again, I do not see the advantage over an explicit loop.

It would be more desirable in my opinion to design `get_quantities_and_times_np()` as a combination of `get_quantities_and_times` plus some additional operations for time t_{n+1}. However, I did not think about all the details.

Reply:
That's why I introduced the iterator version. This way, the same function can be used to fill the vector. I can also use `.insert()` and get rid of the parts you don't like. I am just used to avoid this because it implies more iterations than needed. Since in this part of the code It doesn't matter, I will restructure accordingly.